src/vs/workbench/services/languageRuntime/common/languageRuntimeDataExplorerClient.ts

```diff
@@ -340,7 +356,30 @@ export class DataExplorerClientInstance extends Disposable {
 		profiles: Array<ColumnProfileRequest>
 	): Promise<Array<ColumnProfileResult>> {
 		return this.runBackendTask(
-			() => this._positronDataExplorerComm.getColumnProfiles(profiles, this._profileFormatOptions),
+			async () => {
+				const callbackId = generateUuid();
+				const promise = new DeferredPromise<Array<ColumnProfileResult>>();
+				this._asyncTasks.set(callbackId, promise);
+				await this._positronDataExplorerComm.getColumnProfiles(callbackId, profiles, this._profileFormatOptions);
+
+				const timeout = 10000;
+
+				// Don't leave unfulfilled promise indefinitely; reject after 10 seconds pass
+				// for now just in case
+				setTimeout(() => {
+					// If the promise has already been resolved, do nothing.
+					if (promise.isSettled) {
+						return;
```

Comment:
Do we also need to clean up `_asyncTasks` here?

Reply:
In principle the key should have been deleted from `_asyncTasks` in the resolution of the promise